src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,45 +1,116 @@
-import React from 'react';
+import React, {
+  Dispatch, SetStateAction, useEffect, useState,
+} from 'react';
+
 import { NewCommentForm } from '../NewCommentForm';
 import './PostDetails.scss';
+import { getPostComments } from '../../api/comments';
+import { getPostDetails } from '../../api/posts';
+import { BASE_URL } from '../../api/api';
+
+type Props = {
+  postId: number,
+  onCommentsSet: Dispatch<SetStateAction<Comment[]>>,
+  comments: Comment[],
+};
+
+export const PostDetails: React.FC<Props> = ({
+  postId,
+  onCommentsSet,
+  comments,
+}) => {
+  const [isComment, setIsComment] = useState(false);
+  const [post, setPost] = useState<Post | null>(null);
+
+  const fetchPosts = async () => {
+    const posts = await getPostDetails(postId);
+
+    setPost(posts);
+  };
+
+  const fetchComments = async () => {
+    const commentsArr = await getPostComments(postId);
+
+    onCommentsSet(commentsArr);
+  };
+
+  useEffect(() => {
+    fetchPosts();
+    fetchComments();
+  }, [postId]);
+
+  const removeComment = async (commentId: number) => {
+    const response = await fetch(`${BASE_URL}/comments/${commentId}`,
+      { method: 'DELETE' });
+
+    const data = await response.json();
+
+    return data;
+  };
+
+  return (
+    <div className="PostDetails">
+      <h2>Post details:</h2>
+
+      <section className="PostDetails__post">
+        <p>{post?.title}</p>
+      </section>
+
+      {isComment ? (
+        <button
+          type="button"
+          className="button"
+          onClick={() => {
+            setIsComment(prev => !prev);
+          }}
+        >
+          Hide
+          {' '}
+          {comments.length}
+          {' '}
+          {comments.length > 1 ? 'comments' : 'comment'}
+        </button>
+      ) : (
+        <button
+          type="button"
+          className="button"
+          onClick={() => setIsComment(prev => !prev)}
+        >
+          Show
+          {' '}
+          {comments.length}
+          {' '}
+          {comments.length > 1 ? 'comments' : 'comment'}
+        </button>
+      )}
```

Comment:
These buttons differs only by `Show/Hide` words, don't need to wrap entire buttons into ternary operator